tools/shared/src/main/scala/org/scalajs/core/tools/javascript/Trees.scala

```diff
@@ -21,6 +21,55 @@ import ir.Position.NoPosition
 object Trees {
   import ir.Trees.requireValidIdent
 
+  /** Tests whether a string is a valid ECMAScript `IdentifierName`.
+   *
+   *  `IdentifierName` is defined in
+   *  [[http://www.ecma-international.org/ecma-262/6.0/#sec-names-and-keywords
+   *  Section 11.6 of the ECMAScript 2015 specification]].
+   *
+   *  Currently, this implementation is buggy in some corner cases, as it does
+   *  not accept code points with the Unicode properties `Other_ID_Start` and
+   *  `Other_ID_Continue`. For example,
+   *  `isValidIdentifierName(0x2118.toChar.toString)` will return `false`
+   *  instead of `true`.
+   *
+   *  In theory, it does not really account for code points with the Unicode
+   *  properties `Pattern_Syntax` and `Pattern_White_Space`, which should be
+   *  rejected. However, with the current version of Unicode (9.0.0), there
+   *  seems to be no such character that would be accepted by this method.
+   */
+  final def isValidIdentifierName(name: String): Boolean = {
```

Comment:
Now this can be private, no?

Reply:
It technically can, but I would leave it public. A user of the API would potentially want to call this method to validate inputs, rather than crashing when trying to construct an `ExportName`.

Maybe expose it as `ExportName.isValidExportName` instead?

Reply:
That's fair. Yes I like that name better even if it matches less ES terminology. But in terms of our interface it is definitely cleaner.